### PR TITLE
IncludedBuild migration

### PR DESCRIPTION
**Pull request by CherryPerry**
`buildSrc` has a couple of issues, let's migrate it to [composite build](https://docs.gradle.org/current/userguide/composite_builds.html).

### CODE REVIEW

File: includedBuild/gradleConfiguration/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+dependencies {
+    implementation(Deps.kotlin.plugin)
+    implementation(Deps.android.plugin)
+    implementation(Deps.jmh.plugin)
+    implementation(Deps.bintray)
+    implementation(Deps.detekt.plugin)
+    implementation(Deps.shadow)
+    implementation(Deps.kotlinx.compatibility)
+    // Does not work in IDEA
+    // implementation("com.badoo.reaktive.dependencies:dependencies:SNAPSHOT")
```
Comment (arkivanov) [inline includedBuild/gradleConfiguration/build.gradle.kts:24]: Is it ok?

File: includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/configuration/MppConfigurationExtension.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.badoo.reaktive.configuration
+
+import org.gradle.api.Project
+import javax.inject.Inject
+
+open class MppConfigurationExtension @Inject constructor(
+    private val project: Project
+) {
+    var isLinuxArm32HfpEnabled: Boolean = false
```
Comment (arkivanov) [inline includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/configuration/MppConfigurationExtension.kt:9]: Maybe it's better to make it opt-out?

**Comment by arkivanov** [comment]
Could you please also describe benefits of composite builds for Reaktive project?

File: includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/configuration/MppConfigurationExtension.kt (hunk 2 above)
Comment (CherryPerry) [reply includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/configuration/MppConfigurationExtension.kt:9]: Check article :)

**Comment by arkivanov** [comment]
Feel free to merge if it's ready

**Pull request merged**